bin/pkg/lock.ml

```diff
@@ -5,6 +5,66 @@ module Opam_repo = Dune_pkg.Opam_repo
 module Lock_dir = Dune_pkg.Lock_dir
 module Pin_stanza = Dune_pkg.Pin_stanza
 
+module Progress_indicator = struct
+  module Per_lockdir = struct
+    module State = struct
+      module Repository = Dune_pkg.Pkg_workspace.Repository
+
+      type t =
```

Comment:
To me it would make more sense if these were more aligned with the solving process. `Initializing` is not a step of the solver, it's just a dummy value to put into the state `ref` (it is never assigned outside of initialization).

Will the user ever see `Done`?

Reply:
Users will probably never see `Initializing` or `Done`. I'll update `State.t` to just represent the updating and solving steps, and make the state optional so a `None` value can represent the state before and after locking.